src/NinjaTools.FluentMockServer.API/Proxy/Visitors/ComparasionVisitor.cs

```diff
@@ -0,0 +1,169 @@
+            Score += Visit(visitable.Path);
+            Score += Visit(visitable.Method);
+            Score += Visit(visitable.BodyMatcher);
+            return Score;
+        }
+
+        /// <inheritdoc />
+        public double Visit(RequestBodyMatcher? visitable, CancellationToken token = default)
+        {
+            token.ThrowIfCancellationRequested();
+            if (!visitable?.IsEnabled() ?? true)
+                return Pass();
+
+            Guard.Against.Null(visitable, nameof(visitable) + nameof(RequestBodyMatcher));
+            if (HttpRequest.Body is null)
+                return Fail();
+
+            HttpRequest.EnableBuffering();
+            HttpRequest.Body.Seek(0, SeekOrigin.Begin);
+            using var reader = new StreamReader(HttpRequest.Body);
+            var httpBody = reader.ReadToEnd();
+
+            if (visitable.MatchExact.HasValue && visitable.MatchExact.Value  && httpBody == visitable.Content)
+                return Pass();
+
+            return Fail();
+        }
+
+        /// <inheritdoc />
+        public double Visit(HeaderCollection? visitable, CancellationToken token = default)
+        {
+            token.ThrowIfCancellationRequested();
+
+            if (!visitable?.IsEnabled() ?? true)
+                return Pass();
+
+            Guard.Against.NullOrEmpty(visitable.Headers, nameof(visitable) + nameof(HeaderCollection));
+
+            if (HttpRequest.Headers is null && visitable.Headers.EnsureNotNullNotEmpty() is { })
+                return Fail();
+
+            var success = visitable.Headers.Where(v =>
+            {
+                if (!HttpContext.Request.Headers.TryGetValue(v.Key, out var value))
+                    return false;
+
+                return !value.Except(v.Value).Any();
+
+            }).Any();
+
+            return success ? Pass() : Fail();
+        }
+
+
+        /// <inheritdoc />
+        public double Visit(CookieCollection? visitable, CancellationToken token = default)
+        {
+            token.ThrowIfCancellationRequested();
+            if (!visitable?.IsEnabled() ?? true)
+                return Pass();
+
+            Guard.Against.NullOrEmpty(visitable, nameof(visitable) + nameof(CookieCollection));
+
+            if (HttpRequest.Cookies is null && visitable is { } v && v.Any())
+                Fail();
+
+            if (visitable.Except(HttpContext.Request.Cookies).EnsureNotNullNotEmpty().Any())
+                return Fail();
+
+            return Pass();
+        }
+
+        /// <inheritdoc />
+        public double Visit(QueryCollectionWrapper visitable, CancellationToken token = default)
+        {
+            token.ThrowIfCancellationRequested();
+            if (!visitable?.IsEnabled() ?? true)
+                return Pass();
+
+            Guard.Against.NullOrEmpty(visitable, nameof(visitable) + nameof(QueryCollectionWrapper));
+            if (!HttpRequest.QueryString.HasValue && visitable.QueryString.HasValue)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4802644)